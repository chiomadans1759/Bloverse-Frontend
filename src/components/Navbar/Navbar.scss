@import 'sass-rem';

.vl {
  height: rem(38px);
  border-left: rem(1px) solid #e8e8e8;
  left: 50%;
  margin: auto rem(20px);
  top: 0;
  font-weight: 100;
}

.nav-logo {
  height: rem(40px)
}

.router-link-exact-active {
  cursor: pointer;
  font-weight: 900;
  color: #073cec !important;
}

nav {
  font-size: rem(14px);
  border-bottom: rem(1px) solid #e6e6e6;
  margin-bottom: rem(49px);
  background-color: #ffffff;
}

.transparent {
  background-color: transparent !important;
  border: none;
}

.white-text {
  a.nav-link {
    color: rgba(255, 255, 255, 0.65);
  }
  a.nav-link:hover {
    color: rgb(255, 255, 255);
  }
  a.router-link-active {
    color: white !important;
  }
  a.router-link-active:hover {
    color: white !important;
  }
}

.alt-button {

  .vl {
    height: rem(38px);
    border-left: rem(1px) solid rgba(255, 255, 255, 0.10) !important;
    left: 50%;
    margin: auto rem(20px);
    top: 0;
    font-weight: 100;
    background-color: rgba(255, 255, 255, 0.10) !important;
  }

  .get-started-link {
    background-color: transparent !important;
    border: rem(1px) solid #ffffff;
    color: #ffffff;
  }

  .get-started-link:hover {
    background-color: #ffffff !important;
    background-image: none;
    border: rem(1px) solid #096DD9;
    border-radius: 0.3rem;
    color: #096DD9;
    padding: rem(7px) rem(16px);
    text-align: center;
    text-decoration: none;
    font-size: rem(14px);
  }

  .publish-link {
    background-color: #ffffff !important;
    background-image: none;
    border: rem(1px) solid #096DD9;
    border-radius: 0.3rem;
    color: #096DD9;
    padding: rem(7px) rem(16px);
    text-align: center;
    text-decoration: none;
    font-size: rem(14px);
  }
}

.dropdown-menu a {
  color: #898989;
}

.dropdown-toggle:after {
  visibility: hidden;
}

.nav-link.dropdown-toggle {
  padding-top: rem(6px);
}

.get-started-link {
  background-color: #ffffff;
  border: rem(1px) solid #096DD9;
  border-radius: 0.3rem;
  color: #096DD9;
  padding: rem(7px) rem(16px);
  text-align: center;
  text-decoration: none;
  font-size: rem(14px);
}

.initials {
  &-link {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    background-color: #FF3434;
    border: rem(1px) solid #FF3434;
    border-radius: 0.3rem;
    color: #ffffff;
    padding: rem(7px) rem(0px);
    padding-left: rem(10px);
    text-align: center;
    text-decoration: none;
    font-size: rem(14px);
  }
  .icon {
    &-profile, &-settings {
      color: #096DD9
    }
    &-power {
      color: #FF3434
    }
  }
}

.publish-link {
  background-image: linear-gradient( to bottom, #2F80ED, #003CE1);
  border: rem(1.5px) solid #096DD9;
  border-radius: 0.3rem;
  color: #ffffff;
  padding: rem(7px) rem(16px);
  text-align: center;
  text-decoration: none;
  font-size: rem(14px);
}

.get-started-link:hover {
  background-image: linear-gradient( to bottom, #2F80ED, #003CE1);
  border: rem(1.5px) solid #096DD9;
  border-radius: 0.3rem;
  color: #ffffff;
  padding: rem(7px) rem(16px);
  text-align: center;
  text-decoration: none;
  font-size: rem(14px);
}

.publish-link:hover {
  background-color: #ffffff !important;
  background-image: none;
  border: rem(1px) solid #096DD9;
  border-radius: 0.3rem;
  color: #096DD9;
  padding: rem(7px) rem(16px);
  text-align: center;
  text-decoration: none;
  font-size: rem(14px);
}

.dropdown-menu:after, .dropdown-menu:before {
	bottom: 100%;
	left: 50%;
	border: solid transparent;
	content: " ";
	height: 0;
	width: 0;
	position: absolute;
	pointer-events: none;
}

.dropdown-menu:after {
	border-color: rgba(255, 255, 255, 0);
	border-bottom-color: #ffffff;
	border-width: rem(10px);
	margin-left: rem(-10px);
}

.dropdown-menu:before {
	border-color: rgba(194, 225, 245, 0);
	border-bottom-color: rgba(18, 38, 63, .1);
	border-width: rem(12px);
	margin-left: rem(-12.5px);
}

.dropdown-toggle #option-icon {
  display: inline-block !important;
  background-color: #bcbcbc;
  width: rem(4px);
  height: rem(4px);
  margin-left: rem(2px);
  border-radius: 50%;
}

.dropdown-menu {
  left: rem(-62px);
  top: rem(45px);
}

#feeds-header {
  .backlink {
    margin: 0 rem(-5px);
    color: #565656;
    font-size: rem(16px);
    .back-icon {
      margin: rem(-6px) rem(10px);
    }
    .back-text {
      margin: 0 rem(5px);
    }
  }
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
  .nav-logo {
    margin: -0.375rem;
  }

  .mobile-view {
    margin-right: rem(-42px);
  }

  .dropdown-mobile-view {
    left: rem(122px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 411px) {
  .dropdown-mobile-view {
    left: rem(131px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 400px) {
  .dropdown-mobile-view {
    left: rem(120px);
    top: rem(65px);
  }
}

@media only screen and (min-width: 414px) {
  .dropdown-mobile-view {
    left: rem(135px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 375px) {
  .nav-logo {
    margin: -0.25rem;
  }

  .mobile-view {
    margin-right: rem(-44px);
  }

  .dropdown-mobile-view {
    left: rem(97px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 350px) {
  .nav-logo {
    margin: rem(-44px);
  }

  .mobile-view {
    margin-right: rem(-34px);
  }

  .dropdown-mobile-view {
    left: rem(24px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 360px) {
  .nav-logo {
    margin: -0.4375rem;
  }

  .mobile-view {
    margin-right: rem(-44px);
  }

  .dropdown-mobile-view {
    left: rem(82px);
    top: rem(65px);
  }
}

@media only screen and (max-width: 320px) {
  .nav-logo {
    margin: -1.3125rem;
  }

  .mobile-view {
    margin-right: rem(-44px);
  }

  .dropdown-mobile-view {
    left: rem(42px);
    top: rem(65px);
  }
}

@media only screen and (min-width: 768px) {
  .logo {
    margin: rem(-31px);
  }

  .mobile-view {
    margin-right: rem(-26px);
  }

  .dropdown-mobile-view {
    left: rem(472px);
    top: rem(65px);
  }
  #feeds-header {
    .backlink {
      margin: 0 rem(16px);
    }
  }
}

@media screen and (max-width: 800px) and (min-width: 320px){
  .navbar-toggler {
    border: none !important;
    color: #ffffff !important;
  }
}

