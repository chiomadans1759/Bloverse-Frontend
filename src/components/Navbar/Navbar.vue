<template>
  <header id="feeds-header">
  <nav class="navbar navbar-expand-lg navbar-light px-5"
    :class="{ 'transparent': isTransparent, 'alt-button': isTransparent }">
    <router-link to="/" class="router-link">
      <img class="nav-logo" v-if="!isTransparent" src="@/assets/Logo.svg" />
      <img class="nav-logo" v-if="isTransparent" src="@/assets/Logo - White.svg" />
    </router-link>
    <div class="form-inline my-2 my-lg-0 navbar-toggler mobile-view" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <span id="option-icon"/>
        <span id="option-icon"/>
        <span id="option-icon"/>
      </a>
      <div class="dropdown-menu dropdown-mobile-view" aria-labelledby="navbarDropdown">
        <router-link to="/about" class="dropdown-item">About</router-link>
        <router-link to="/photocontest" class="dropdown-item">Photo Contest</router-link>
        <router-link to="/terms-and-conditions" class="dropdown-item">Terms</router-link>
        <router-link to="/privacy-policies" class="dropdown-item">Privacy</router-link>
        <a class="dropdown-item" href="#">FAQ</a>
        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="#">Feedback</a>
      </div>
      <div>
         <span class="vl bg-light"></span>
          <router-link class="get-started-link" to="/login">
            Get Started
          </router-link>
      </div>
    </div>

    <div class="collapse navbar-collapse">
      <ul class="navbar-nav ml-auto" :class="{ 'white-text': whiteText }">
        <li class="nav-item active">
          <router-link to="/" class="nav-link"> Home <span class="sr-only">(current)</span></router-link>
        </li>
        <li class="nav-item">
          <router-link to="/about" class="nav-link">About</router-link>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="/photocontest">Photo Contest</a>
        </li>
        <li class="nav-item dropdown pr-4">
          <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
            <span id="option-icon"/>
            <span id="option-icon"/>
            <span id="option-icon"/>
          </a>
          <div class="dropdown-menu" aria-labelledby="navbarDropdown">
            <router-link to="/terms-and-conditions" class="dropdown-item">Terms</router-link>
            <router-link to="/privacy-policies" class="dropdown-item">Privacy</router-link>
            <router-link to="/settings" class="dropdown-item">Settings</router-link>
            <a class="dropdown-item" href="#">FAQ</a>
            <div class="dropdown-divider"></div>
            <a class="dropdown-item" href="#">Feedback</a>
          </div>
        </li>
      </ul>
      <div class="form-inline my-2 my-lg-0 pr-4" :class="{ 'alt-button': isTransparent }">
          <router-link class="get-started-link" to="/login">
            Get Started
          </router-link>
        <span class="vl bg-light"></span>
          <router-link class="publish-link" to="/creators/apply">
            Publish
          </router-link>
      </div>
    </div>
</nav>
  </header>
</template>

<script>
import { Row, Col } from 'iview';
import { mapState, mapActions } from "vuex";

export default {
  components: { Row, Col },
  props: {
    isTransparent: {
      type: Boolean,
      default: false
    },
    whiteText: {
      type: Boolean,
      default: false
    },
  },
  methods: {
    ...mapActions(['clearSession'])
  },
  computed: {
    ...mapState(["auth"])
  }
}
</script>

<style lang="scss" scoped>
  @import './Navbar';
</style>

