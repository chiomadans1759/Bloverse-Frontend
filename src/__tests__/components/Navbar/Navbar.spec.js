import { mount } from '@vue/test-utils';
import Navbar from '@/components/Navbar/Navbar.vue';

describe('Navbar Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(Navbar);

    expect(wrapper.element).toMatchSnapshot();
  });
});