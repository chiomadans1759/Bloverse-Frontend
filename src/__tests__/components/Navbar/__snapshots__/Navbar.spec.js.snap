// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `
<header
  id="feeds-header"
>
  <nav
    class="navbar navbar-expand-lg navbar-light px-5"
  >
    <a
      class="router-link"
    >
      <img
        class="nav-logo"
        src="@/assets/Logo.svg"
      />
       
      <!---->
    </a>
     
    <div
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="form-inline my-2 my-lg-0 navbar-toggler mobile-view"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
    >
      <a
        aria-expanded="false"
        aria-haspopup="true"
        class="nav-link dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="navbarDropdown"
        role="button"
      >
        <span
          id="option-icon"
        />
         
        <span
          id="option-icon"
        />
         
        <span
          id="option-icon"
        />
      </a>
       
      <div
        aria-labelledby="navbarDropdown"
        class="dropdown-menu dropdown-mobile-view"
      >
        <a
          class="dropdown-item"
        >
          About
        </a>
         
        <a
          class="dropdown-item"
        >
          Photo Contest
        </a>
         
        <a
          class="dropdown-item"
        >
          Terms
        </a>
         
        <a
          class="dropdown-item"
        >
          Privacy
        </a>
         
        <a
          class="dropdown-item"
          href="#"
        >
          FAQ
        </a>
         
        <div
          class="dropdown-divider"
        />
         
        <a
          class="dropdown-item"
          href="#"
        >
          Feedback
        </a>
      </div>
       
      <div>
        <span
          class="vl bg-light"
        />
         
        <a
          class="get-started-link"
        >
          
            Get Started
          
        </a>
      </div>
    </div>
     
    <div
      class="collapse navbar-collapse"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item active"
        >
          <a
            class="nav-link"
          >
             Home 
            <span
              class="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
          >
            About
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/photocontest"
          >
            Photo Contest
          </a>
        </li>
         
        <li
          class="nav-item dropdown pr-4"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            <span
              id="option-icon"
            />
             
            <span
              id="option-icon"
            />
             
            <span
              id="option-icon"
            />
          </a>
           
          <div
            aria-labelledby="navbarDropdown"
            class="dropdown-menu"
          >
            <a
              class="dropdown-item"
            >
              Terms
            </a>
             
            <a
              class="dropdown-item"
            >
              Privacy
            </a>
             
            <a
              class="dropdown-item"
              href="#"
            >
              FAQ
            </a>
             
            <div
              class="dropdown-divider"
            />
             
            <a
              class="dropdown-item"
              href="#"
            >
              Feedback
            </a>
          </div>
        </li>
      </ul>
       
      <div
        class="form-inline my-2 my-lg-0 pr-4"
      >
        <a
          class="get-started-link"
        >
          
            Get Started
          
        </a>
         
        <span
          class="vl bg-light"
        />
         
        <a
          class="publish-link"
        >
          
            Publish
          
        </a>
      </div>
    </div>
  </nav>
</header>
`;
